File: services/api/src/models/signal.py
from enum import Enum
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime


class SignalStatus(str, Enum):
    pending = "pending"
    approved = "approved"
    rejected = "rejected"
    expired = "expired"
    executed = "executed"


class SignalCreate(BaseModel):
    strategy: str
    market_id: str
    our_probability: float = Field(..., ge=0.0, le=1.0)
    market_price: float = Field(..., ge=0.0, le=1.0)
    edge: float
    confidence: str = Field(..., pattern="^(high|medium|low)$")
    reasoning: str
    model_breakdown: Optional[dict] = None


class SignalResponse(SignalCreate):
    id: str
    status: SignalStatus = SignalStatus.pending
    created_at: datetime
    approved_at: Optional[datetime] = None
    approved_by: Optional[str] = None
    rejection_reason: Optional[str] = None
    trade_id: Optional[str] = None

File: apps/web/src/types/signal.ts
export type SignalStatus = 'pending' | 'approved' | 'rejected' | 'expired' | 'executed'
export type SignalDirection = 'buy_yes' | 'buy_no' | 'sell_yes' | 'sell_no'

export interface Signal {
  id: string
  market_id: string
  strategy_id: string
  direction: SignalDirection
  our_probability: number
  market_price: number
  edge: number
  confidence: number
  kelly_fraction: number
  recommended_size: number
  status: SignalStatus
  reason: string | null
  market_question: string | null
  strategy_name: string | null
  created_at: string
  updated_at: string
  expires_at: string | null
}
